qiskit_experiments/library/randomized_benchmarking/mirror_rb_experiment.py

```diff
@@ -0,0 +1,434 @@
+        options.update_options(
+            sampling_algorithm="edge_grab",
+            start_end_clifford=True,
+            pauli_randomize=True,
+            two_qubit_gate_density=0.2,
+            two_qubit_gate=CXGate(),
+            sampler_opts={},
+            inverting_pauli_layer=False,
+        )
+        options.set_validator(field="two_qubit_gate_density", validator_value=(0, 1))
+
+        return options
+
+    def circuits(self) -> List[QuantumCircuit]:
+        """Return a list of Mirror RB circuits.
+
+        Returns:
+            A list of :class:`QuantumCircuit`.
+        """
+        sequences = self._sample_sequences()
+        circuits = self._sequences_to_circuits(sequences)
+
+        return circuits
+
+    def _set_distribution_options(self):
+        """Set the coupling map and gate distribution of the sampler
+        based on experiment options. This method is currently implemented
+        for the default "edge_grab" sampler."""
+
+        if self.experiment_options.sampling_algorithm != "edge_grab":
+            raise QiskitError(
+                "Unsupported sampling algorithm provided. You must implement"
+                "a custom `_set_distribution_options` method."
+            )
+
+        self._distribution.seed = self.experiment_options.seed
+
+        # Coupling map is full connectivity by default. If backend has a coupling map,
+        # get backend coupling map and create coupling map for physical qubits converted
+        # to qubits 0, 1...n
+        if self.backend and self._backend_data.coupling_map:
+            coupling_map = CouplingMap(self._backend_data.coupling_map)
+        else:
+            coupling_map = CouplingMap.from_full(len(self.physical_qubits))
+
+        self._distribution.coupling_map = coupling_map.reduce(self.physical_qubits)
+
+        # Adjust the density based on whether the pauli layers are in
+        if self.experiment_options.pauli_randomize:
+            adjusted_2q_density = self.experiment_options.two_qubit_gate_density * 2
+        else:
+            adjusted_2q_density = self.experiment_options.two_qubit_gate_density
+
+        if adjusted_2q_density > 1:
+            warnings.warn("Two-qubit gate density is too high, capping at 1.")
+            adjusted_2q_density = 1
+
+        if self.experiment_options.pauli_randomize:
+            self._distribution.gate_distribution = [
+                GateDistribution(
+                    prob=adjusted_2q_density, op=self.experiment_options.two_qubit_gate
+                ),
+                GateDistribution(prob=1 - adjusted_2q_density, op=GenericClifford(1)),
+            ]
+
+        else:
+            self._distribution.gate_distribution = [
+                GateDistribution(
+                    prob=adjusted_2q_density, op=self.experiment_options.two_qubit_gate
+                ),
+                GateDistribution(prob=1 - adjusted_2q_density, op=GenericClifford(1)),
+            ]
```

Comment:
No difference in `if` and `else` block?

Reply:
Thanks for catching this, removed duplicate code.